R/translator.R

```diff
@@ -0,0 +1,100 @@
+#' Translator options
+.translator_options <- list(
+  cultural_bignumer_mark = NULL,
+  cultural_punctuation_mark = NULL,
+  cultural_date_format = NULL
+)
+
+#' Translator class
+#'
+#' @field languages character.
+#' @field key_translation character.
+#' @field translation_file character.
+#' @field options list.
+#' @field translations data.frame.
+#' @field translation_language character.
+#'
+#' @return
+#' @export
+#'
+#' @import jsonlite
+#'
+#' @examples
+#' @rdname translator
+Translator <- setRefClass(
+  "Translator",
+  fields = list(
+    languages = "character",
+    key_translation = "character",
+    translation_language = "character",
+    translation_file = "character",
+    options = "list",
+    translations = "data.frame"
+  )
+)
+
+#' @rdname translator
+Translator$methods(
+  initialize = function(translation_file, key_translation) {
+    translation_file <<- translation_file
+    options <<- .translator_options
+    # TODO validate format of translation_file
+    # Update the list of options, or take a default from config.
+    json_data <- jsonlite::fromJSON(translation_file)
+    common_fields <- intersect(names(json_data), names(options))
+    options <<- modifyList(options, json_data[common_fields])
+
+    # If there is no English in translations, pick the first one from the list.
```

Comment:
Outdated comment: that was my first idea before I realise that we can use LANG env var.